### PR TITLE
Mitigate Azure object store stragglers by limiting gateway connections

### PR DESCRIPTION
**Pull request by parasj**
Past 24 connections, Azure blob or Azure auth begins to throttle performance. This PR sets a default limit of 24 connections for Azure.

### CODE REVIEW

**Review by ShishirPatil — COMMENTED** [review]
Chunking is an artifact introduced by skyplane, and chunk # is not an underlying object (thought here is a 1-1 mapping). This information is critical for debugging, but is not relevant to the user. If I know there are 3 chunks remaining that might be Ok, though I don't see how that would be useful beyond informatics. The additional chunk id in-itself maybe is not so useful. 

Is there any reason why you thought it would be good to print this out?

File: skyplane/replicate/replicator_client.py [hunk 1]
```diff
@@ -609,11 +609,16 @@ def monitor_transfer(
                         # update progress bar
                         total_runtime_s = (log_df.time.max() - log_df.time.min()).total_seconds()
                         throughput_gbits = completed_bytes * 8 / GB / total_runtime_s if total_runtime_s > 0 else 0.0
+                        pending_chunk_ids = set([cr.chunk.chunk_id for cr in job.chunk_requests]) - set(completed_chunk_ids)
+                        if len(pending_chunk_ids) < 5:
+                            pending_chunks = ", chunks [" + ", ".join([str(cid) for cid in pending_chunk_ids]) + "] remain"
```
Comment (ShishirPatil) [inline skyplane/replicate/replicator_client.py:614]: May not be necessary, re comment

Comment (parasj) [reply skyplane/replicate/replicator_client.py]: I'll only enable this if `debug` is passed.

**Comment by parasj** [comment]
@ShishirPatil These stats were useful for debugging stragglers, but I agree it shouldn't be user facing. I gated off that logic with a `debug` flag.

**Pull request merged**